### DISCUSSION TITLE
[JSX] Do not add the import statement if source is not provided

### DISCUSSION THREAD

**Discussion by lkenny97** [mdx-editor/editor#50] [Ideas]
Hello,

I am using `next-mdx-remote` to display my content generated by `mdx-editor` and stored remotely.

According to the docs, `next-mdx-remote` does not use the import statement and takes the components as a prop. [See docs.](https://github.com/hashicorp/next-mdx-remote#import--export)

Currently, I am thinking of just deleting first 2 lines of the generated content before storing it in order not to have an import statement in my content.

Here is QoL feature that would be helpful as next-mdx-remote is becoming the go to repo.

```
const jsxComponentDescriptors: JsxComponentDescriptor[] = [
  {
    name: 'MyLeaf',
    kind: 'text', 

    // Leaving this undefined or null will not generate the import statement on top of the file
    source: undefined,

    props: [
      { name: 'foo', type: 'string' },
      { name: 'bar', type: 'string' }
    ],
    hasChildren: true,
    Editor: GenericJsxEditor
  },
]
```

Leaving it undefined in current version just adds
`import { MyLeaf } from 'undefined'`

**Comment by petyosi** [comment]
Oh, that's great, thank you for sharing. Should be an easy enhancement, will look into it. Converting this to a feature "issue".

**Comment by petyosi** [comment]
Just pushed a fix for that in v0.17.2. Closing the discussion, but let me know if there's more to it.
